tensorboard/plugins/debugger/README.md

```diff
@@ -0,0 +1,144 @@
+
+## The Node List
+
+Ops can be selected via the the node list on the top left:
+
+![Node list](images/node_list.png)
+
+Toggling a checkbox next to an entire scope selects or deselects all ops under the scope. The checkbox for a scope is orange if some but not all of the ops within it are selected.
+
+The ops shown in the list may be filtered by regex based on name or type. Afterwards, ops may be more efficiently selected by the user.
+
+![Select ops by regex](images/regex_node_list.png)
+
+Clicking the link next to an op makes the graph explorer pan to it (and expand nodes if need be to show the op):
+
+![Make the graph explorer pan to a node](images/node_linking.png)
+
+## The Graph Explorer
+
+The graph explorer (on the right side of the dashboard) offers another way to select an op: A context menu appears when an op is right-clicked.
+
+The user can then chose to either
+
+1. Set a breakpoint at the op (equivalent to selecting in the node list).
+2. Continue to the op. This convenient option sets the breakpoint and then continues execution to the op.
+
+![Add a breakpoint via the graph explorer](images/context_menu.png)
+
+# Controlling Execution
+
+After selecting nodes, the user can continue execution for a certain number of session runs by clicking *Continue*:
+
+![The continue button](images/continue_button.png)
+
+Clicking the button opens a dialog that lets the user specify how many session runs to execute. Execution will pause at breakpoints (selected ops).
+
+![The continue dialog](images/continue_dialog.png)
+
+Via this dialog, the user may specify breakpoints based on conditions (in addition to the breakpoints that are based on selected ops):
+
+* When a tensor contains any bad (NaN, or +/- Infinity) values.
+* When a tensor contains any +/- Infinity values.
+* When a tensor contains any NaN values.
+* When the max value of a tensor exceeds some constant.
+* When the max value of a tensor is below some constant.
+* When the min value of a tensor exceeds some constant.
+* When the min value of a tensor is below some constant.
+* When the (max - min) value of a tensor exceeds some constant.
+* When the (max - min) value of a tensor is below some constant.
+* When the mean value of a tensor exceeds some constant.
+* When the mean value of a tensor is below some constant.
+* When the standard deviation of a tensor exceeds some constant.
+* When the standard deviation of a tensor is below some constant.
+
+These conditions bear much semblance to [filters of the TensorFlow debugger](https://www.tensorflow.org/programmers_guide/debugger#finding_nans_and_infs).
+
+When execution is paused, the next op can be stepped to via clicking *Step*. If a program runs multiple sessions, they will be listed within the Session Runs table under the node list.
+
+![Step to the next op](images/step.png)
+
+# Examining Runtime Graphs
```

Comment:
This section seems to be somewhat redundant given the "## The Graph Explorer" section above. Please remove it or combine them.

Reply:
Done.